### ISSUE TITLE
MarianMTModel is not trained after finished training and loading

### ISSUE THREAD

**Issue by marekninja** [huggingface/transformers#16048]
## Environment info

- `transformers` version: 4.17.0.dev0
- Platform: Linux-5.4.0-92-generic-x86_64-with-glibc2.29
- Python version: 3.8.10
- PyTorch version (GPU?): 1.12.0.dev20220208+cu102 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help

@patrickvonplaten

## Information

I am training MarianMTModel for translation from scratch on EuroParl dataset. 
* validation BLEU during training is cca 30, checkpointing every 5000 steps, with load_best_model_at_end argument
* At the end of the training I save the model with `trainer.save_model('./trained')`.
* I then load this model with `MarianMTModel.from_pretrained("./trained")`. Loaded model does not appear to be trained, BLEU score on validation set is around 4
* Came around this: 
https://github.com/huggingface/transformers/blob/e66743e6c9601a4b12ffc2335a9f60a41d1ca60c/src/transformers/models/marian/modeling_marian.py#L1211-L1224
* Finally, solved the issue by setting `model._keys_to_ignore_on_save = None`

I believe the problem is exactly the same as in issue #9525

I do not know about `_keys_to_ignore_on_load_missing`, but I think the `_keys_to_ignore_on_save` should be None.

My config also has `"static_position_embeddings": true,`, but still it seems like, embedings are trained too...

## To reproduce

1. Train MarianMTModel from scratch
2. Save checkpoint/best model with `trainer.save_model('./trained')`
3. Load checkpoint with from_pretrained
4. BLEU score is not the same (it is much lower)

## Expected behavior

Model should also save weights of positional  #embeddings.

**Comment by marekninja** [comment]
Found out the real issue. I was resetting the model incorrectly... I did reset_parameters on all layers (including positional embeddings), but in combination with static positional embeddings it made the model unusable after loading. My bad.
